### DISCUSSION TITLE
NRF24L01+ and CMT2300A simultaneously on one DTU, manage HM and HMS inverters with one DTU

### DISCUSSION THREAD

**Discussion by bennoulli** [tbnobody/OpenDTU#1753] [Ideas]
Hi Community,

EN: I am thinking about managing an HM-1500 and an HMS-2000 with the same openDTU device. Are there tested ways to connect two RF modules (NRF24L01+ and CMT2300A) simultaneously, and is there a code base to operate the two modules simultanesously?

DE: Ich würde gerne an einer DTU sowohl einen HM-1500 als auch einen HMS-2000 überwachen. Gibt es schon irgendwo Ansätze, an einem ESP32 zwei Funkmodule zu betreiben, ein NRF24L01+ und gleichzeitig ein CMT2300A?

Falls nicht, würde ich zur Diskussion stellen wollen, ob eine solche Funktionalität von vielen Nutzern gewünscht wäre.
(und übrigens: als "nicht-Programmierer" habe ich großen Respekt vor der Leistung der Developer, die hier beitragen).

Euer Bernoulli

**Comment by ms49434** [comment]
Das Fusion Board hat beide Funkmodule: https://shop.allianceapps.io/products/opendtu-fusion-community-edition

**Comment by Marc9485** [comment]
Hab ich mir dort zugelegt. Antwortet zeitig auf Fragen, schneller Versand. KNn ich nur empfehlen 

https://www.kleinanzeigen.de/s-anzeige/opendtu-auf-platine-in-gehaeuse-balkonkraftwerk-hoymiles-ahoydtu/2642579592-168-2140?utm_source=copyToPasteboard&utm_campaign=socialbuttons&utm_medium=social&utm_content=app_ios

**Reply by polarbearing** [reply]
Der Link scheint tot zu sein?

**Reply by Marc9485** [reply]
Mittlerweile ja

**Reply by ghost** [reply]
Hi,
[Alternativer Artikel](https://www.ebay.de/itm/204669978739?_trkparms=amclksrc%3DITM%26aid%3D1110025%26algo%3DHOMESPLICE.COMPOSITELISTINGS%26ao%3D1%26asc%3D20220405142716%26meid%3Dbe8ee1594de540e6b41c8fc3cb46e4f8%26pid%3D101506%26rk%3D8%26rkt%3D25%26sd%3D186022131763%26itm%3D204669978739%26pmt%3D1%26noa%3D1%26pg%3D4247815%26algv%3DAlgoIndex5SimRanker&_trksid=p4247815.c101506.m1851)

**Comment by polarbearing** [comment]
As I have a running module with NRF24L01+ my question was similar, if I can retrofit the CMT2300A the existing CPU. 
So far I found in the [manuals](https://www.opendtu.solar/hardware/cmt2300a/):
"If you want to run the CMT2300A module on the same ESP32 as a NRF24L01+ module ... make sure you get a ESP which supports 2 SPI busses." and they can use different GPIO pins: "The connection pins from the CMT2300A module to the ESP32 module can dynamically configured using [Device Profiles](https://www.opendtu.solar/firmware/device_profiles/) (prefered method) or at compile time."

Haven't tested it yet, just planning the second module.  My CPU is an ESP32 NodeMCU dev board from Berrybase, so I need to check if that fulfills the 2 SPI condition. Apparently that fails: "Der NodeMCU hat 38 Pins, darunter GPIO-Pins, ein ADC-Pin, ein PWM-Pin, einen SPI-Bus, einen I2C-Bus, eine UART-Schnittstelle..." So 1 SPI only?!

**Reply by ghost** [reply]
Hi,
Die Beschreibung von Berrybase ist irreführend.
Der ESP-WROOM-32 müsste 4 SPI Schnittstellen haben, von denen du 2 benutzen kannst.
Sprich, du kannst beide Funkmodule anschließen und openDTU drauf laden.

**Comment by Fokker-1** [comment]
Hi,

ich habe diese Fusion OpenDTU mit HMS-1000 und HM-800 (mit den 3 externen Antennen), geht prima:

https://3printd-solution.eu/
